Add passive insulting

AllemonIsGit opened the pull request:
Up for review

AllemonIsGit committed:
add exclude user from random insults

AllemonIsGit committed:
add InsultListener

AllemonIsGit committed:
update insulting
now InsultListener responsible for all insult related tasks

novitzkee commented on src/main/java/org/example/listener/InsultListener.java:
```diff
@@ -0,0 +1,65 @@
+import java.util.Objects;
+import java.util.Random;
+
+public class InsultListener extends ListenerAdapter {
```
Why separate listener ? It's logic looks like a duplicate of GlobalListener and additional method is now needed in SlashCommandManager `org.example.manager.SlashCommandManager#getInsultSlashCommandsData`. If splitting listeners is really required, I would think of a way to avoid such code duplication (maybe using an abstract class).

AllemonIsGit replied:
\+ and addressed.
Listeners are separate because their responsibility is different.
They now inherit from parent abstract class "Listener" to reduce code duplication.

novitzkee commented on src/main/java/org/example/listener/InsultListener.java:
```diff
@@ -0,0 +1,65 @@
+
+    public InsultListener() {
+        this.slashCommandManager = SlashCommandManager.getInstance();
+        slashCommands = slashCommandManager.getInsultSlashCommands();
```
Are fields like this really needed ? When you need the commands you can get them from manager, I think fields like this are redundant.

AllemonIsGit replied:
+ and addressed

AllemonIsGit replied:
Due to structure changes those fields are no longer redundant.

novitzkee commented on src/main/java/org/example/listener/InsultListener.java:
```diff
@@ -0,0 +1,65 @@
+    public void onMessageReceived(MessageReceivedEvent event) {
+        List<String> splitMessage = eventService.splitMessageOnSpace(eventService.getMessageFromEvent(event));
+
+        if (!eventService.checkForPrefix(splitMessage.get(0)) && !event.getAuthor().isBot()) {
```
Those methods from EventService don't make much sense. What we are really interested in is getting invoke phrase from event message, why not put that action in single method ?

AllemonIsGit replied:
\+ and addressed.
EventService now has a HandleMessageEvent method.

novitzkee commented on src/main/java/org/example/listener/InsultListener.java:
```diff
@@ -0,0 +1,65 @@
+        String command = event.getFullCommandName();
+        for (SlashCommand slashCommand : slashCommands) {
+            if (Objects.equals(command, slashCommand.getData().getName())) {
+                slashCommand.execute(event);
```
Essentially command is retrieved by name and executed, why not create method for getting it by name in SlashCommandManager ?

AllemonIsGit replied:
This became difficult, due to structural changes, since both Listeners inherit from the same parent, there are problems deciding which set of commands should be checked causing both to execute the command.

Replaced with stream for now, waiting for further suggestions.

Edit.
What I meant was, both classes that inherit from Listener have different sets of commands, with parent class cannot differentiate.

novitzkee commented on src/main/java/org/example/slashcommand/SlashConsent.java:
```diff
@@ -0,0 +1,45 @@
+    private String description = "Toggles whether or not you want to be insulted be the bot.";
+    private CommandData data;
+    private String optionName = "consent";
+    private String optionDescription = "Do you want to be insulted or not?";
```
Fields above should probably be made private static final.

AllemonIsGit replied:
\+ and addressed, all String fields of SlashCommands and TextCommands set to private static final.

novitzkee commented on src/main/java/org/example/manager/SlashCommandManager.java:
```diff
@@ -12,38 +11,55 @@
+        for (SlashCommand slashCommand : genericSlashCommands) {
             data.add(slashCommand.getData());
         }
         return data;
```
Loop here could be replaced with stream, and is a duplicated with one below.

AllemonIsGit replied:
\+ and addressed, replaced with ArrayList::forEach for both cases.

novitzkee replied:
I would see it more like this.

```java
    public List<CommandData> getGenericSlashCommandsData() {
        return getCommandData(genericSlashCommands);
    }

    public List<CommandData> getInsultSlashCommandsData() {
        return getCommandData(insultSlashCommands);
    }

    private List<CommandData> getCommandData(List<SlashCommand> slashCommands) {
        return slashCommands.stream()
                .map(SlashCommand::getData)
                .collect(Collectors.toList());
    }
```

novitzkee commented on src/main/java/org/example/domain/model/GuildUser.java:
```diff
@@ -17,6 +17,8 @@ public class GuildUser {
     @Column(nullable = false)
     private Integer points;
+    @Column(nullable = false)
+    private Boolean insultConsent = false;
```
Why not separate fields with newlines ?

AllemonIsGit replied:
There will be less code visible on screen.
Why separate fields with newlines?

novitzkee replied:
It's more conventional. If someone is used to it, eyes hurt a bit seeing it like this. It's only cosmetic so I'm ok with it from now on.

novitzkee commented on src/main/java/org/example/slashcommand/SlashConsent.java:
```diff
@@ -0,0 +1,45 @@
+            guildUser.setInsultConsent(false);
+            guildUserService.update(guildUser);
+            event.reply("I'm sorry if i offended you, i won't do it from now on.").setEphemeral(true).queue();
+        }
```
Primitive boolean can be used here. Also logic only branches when replying to event, maybe only this action should be placed in if-else block.

AllemonIsGit replied:
\+ and addressed.

novitzkee reviewed (changes requested):
I recommend to consider adding some tests.

AllemonIsGit committed:
Removed redundant code

AllemonIsGit committed:
Code improvements according to @novitzkee suggestions
- Listeners are now inheriting from Listener abstract class to reduce code duplication.
- EventService responsibility for handling message events now closed off to one method.

AllemonIsGit committed:
Code improvements according to @novitzkee suggestions
Listener simplified

AllemonIsGit committed:
Code improvements according to @novitzkee suggestions
Adjusted String fields of SlashCommand and TextCommand

AllemonIsGit committed:
Code improvements according to @novitzkee suggestions
- simplified code in SlashCommandManager

AllemonIsGit committed:
Code improvements according to @novitzkee suggestions
- simplified code in SlashConsent

AllemonIsGit commented:
@novitzkee Most methods use events send by discord, how do I test that?

novitzkee commented on src/main/java/org/example/slashcommand/SlashDeleteInsult.java:
```diff
@@ -8,11 +8,11 @@
     private CommandData data;
-    private String optionName = "id";
-    private String optionDescription = "Provide ID of the insult you want to delete.";
+    private static final String optionName = "id";
```
Putting static fields above non-static ones would be more conventional.
